streaming/storage/deque.go

```diff
@@ -265,6 +265,25 @@ func (dq *Deque) Clear() error {
 	return nil
 }
 
+func (dq *Deque) Length() (int, error) {
```

Comment:
Just a comment - since this calls both initializeBack() and initializeFront() it's not thread safe I think?